hail/python/hail/ir/ir.py

```diff
@@ -56,7 +56,7 @@ class I32(IR):
     @typecheck_method(x=int)
     def __init__(self, x):
         super().__init__()
-        self.x = x
+        self.x = int(x)
```

Comment:
This conversion is because if we pass an `IntEnum` to the IR constructors, it will generally be assigned as is. When rendered, we (generally) use format strings (`__format__` method, which generally delegates to the `__str__` method) to render these items. I would like the `__str__` method of `AlleleType` to be the string that `allele_type` would return, but then when used as a number, we need to make sure to _actually convert it to a number_. 

I'm not sure what to do here. Let's discuss.

Reply:
Seems to me the root of the problem is you've made `AlleleType` a subtype of `int`, but its `str` doesn't act like an int. I'd say it should either fully quack like an int, or you should be forced to call `x.value` to get the int representation.

If we instead forced calling `x.name` to get the string name, where would that actually impact users?

Reply:
(Note that the default `str` of `IntEnum` doesn't return the int representation either, but they fixed that in 3.11, so we can just override it to match the future behavior)

Reply:
Thanks for the commentary. I'm going add a `pretty_name` that matches with the result of `allele_type`.

Reply:
And now we're back to no changes to ir.py